dogfood.md

```diff
@@ -0,0 +1,167 @@
+  createdOn: String!
+  updatedOn: String!
+  photoUrl: String!
+  comments: [Comment!]!
+}
+
+type Comment {
+  id: ID! @unique
+  ownerId: ID!
+  createdOn: String!
+  updatedOn: String!
+  comment: String!
+}
+
+type Query {
+  profile(email: String!):Profile!
+  usersMemes(email: String!): [Meme!]!
+  feed(memeListId: ID!): [Meme!]!
+  meme(memeId: ID!):Meme
+}
+
+type Mutation {
+  createProfile(email: String!, display_name: String!, biography: String!, pictureUrl: String!):Profile!
+  updateProfile(id: ID!, email: String!, display_name: String!, biography: String!, pictureUrl: String!):Profile!
+  deleteProfile(id: ID!):Profile!
+
+  postMeme(picture: [Int!]!, topComment: String!,bottomComment: String!, memeListId: ID!): Meme!
+  deleteMeme(id: ID!): Meme!
+  likeMeme(id: ID!): Meme!
+  unlikeMeme(id: ID!): Meme!
+
+  postComment(memeId: ID!, comment: String!): Meme!
+  deleteComment(id: ID!): Meme!
+}
+```
+
+## Mobile-Core features used
+
+Memeolist will be build using technologies provided with the mobile-core service catalog.  Additional services will be created as necessary using RHOAR runtimes and exposed via a custom connector.  Finally, we will use this as an opportunity to test graphql style APIs which will serve as the basis for the future sync project.
+
+### Custom Service Connector
+
+The final meme images will be generated server side using a RHOAR runtime (vert.x, node.js, or Wildfly Swarm) and exposed using the custom service connector feature.  This means that our SDKs will be responsible for upload the image, the author metadata, and the message to be displayed on it and the RHOAR service will compose that data into a Meme post and save it to a persistent data store.
+
+### Keycloak
+
+Social integration will be provided with KeyCloak and we will use our auth SDKs.  We should also provide a templated realm and a script so that community users can plug in their developer credentials if they wish to host their own version of memeolist.
+
+### Push
+
+We should use UPS to notify users of activity on memes they have published or liked.
+
+### Metrics
+
+Our metrics SDK should be used to collect the following information 
+ * Platform start-up and device information 
+ * Errors and exceptions that are happen
+ * Length of time the application is open 
+ * Number of memes a user interact with during a session broken fown by views, comments, and likes
+
+### Sync / GraphQL PoC
+
+As part of dogfooding we should also try to use this as an opportunity to upskill on GraphQL and see how GraphQL technologies integrate with our current services.  This will be used to inform decisions to be made regarding how the future sync product will be developed.
+
+Of special note is to see how well the subscription feature of GraphQL can be used to create a realtime experience, the caching features can be used to reduce network overhead, and how we can use UPS to trigger updates to the local data when the application is backgrounded.
```

Comment:
@darahayes @aliok This hits on 'Subscription' support, which is outlined in the sync proposal,
and there's a card for that
https://trello.com/c/DL0lV5I0/80-13-as-a-developer-i-want-the-ability-to-synchronise-non-binary-data-in-realtime-using-the-data-sync-service-such-that-all-client

@danielpassos the Apollo Client has Subscription support, so the main ask here is how to use it in the Memeolist app,
and get it to work with the data sync server